Providers/Profile.swift

```diff
@@ -502,48 +498,44 @@ public class BrowserProfile: Profile {
         self.remoteClientsAndTabs.insertCommands(commands, forClients: clients) >>> { self.syncManager.syncClients() }
     }
 
-    lazy var logins: protocol<BrowserLogins, SyncableLogins, ResettableSyncStorage> = {
+    lazy var logins: BrowserLogins & SyncableLogins & ResettableSyncStorage = {
         return SQLiteLogins(db: self.loginsDB)
     }()
 
     // This is currently only used within the dispatch_once block in loginsDB, so we don't
     // have to worry about races giving us two keys. But if this were ever to be used
     // elsewhere, it'd be unsafe, so we wrap this in a dispatch_once, too.
-    private var loginsKey: String? {
+    lazy private var loginsKey: String? = {
         let key = "sqlcipher.key.logins.db"
-        struct Singleton {
-            static var token: dispatch_once_t = 0
-            static var instance: String!
-        }
-        dispatch_once(&Singleton.token) {
-            if self.keychain.hasValueForKey(key) {
-                let value = self.keychain.stringForKey(key)
-                Singleton.instance = value
-            } else {
-                let Length: UInt = 256
-                let secret = Bytes.generateRandomBytes(Length).base64EncodedString
-                self.keychain.setString(secret, forKey: key)
-                Singleton.instance = secret
-            }
-        }
-        return Singleton.instance
-    }
+        return key
+//
+//        struct Singleton {
+//            static var token: dispatch_once_t = 0
+//            static var instance: String!
+//        }
+//        dispatch_once(&Singleton.token) {
+//            if self.keychain.hasValueForKey(key) {
+//                let value = self.keychain.stringForKey(key)
+//                Singleton.instance = value
+//            } else {
+//                let Length: UInt = 256
+//                let secret = Bytes.generateRandomBytes(Length).base64EncodedString
+//                self.keychain.setString(secret, forKey: key)
+//                Singleton.instance = secret
+//            }
+//        }
+//        return Singleton.instance
+    }()
 
     private static var loginsDBCreated = false
     private lazy var loginsDB: BrowserDB = {
-        struct Singleton {
-            static var token: dispatch_once_t = 0
-            static var instance: BrowserDB!
-        }
-        dispatch_once(&Singleton.token) {
-            Singleton.instance = BrowserDB(filename: "logins.db", secretKey: self.loginsKey, files: self.files)
-            BrowserProfile.loginsDBCreated = true
-        }
-        return Singleton.instance
+        let db = BrowserDB(filename: "logins.db", secretKey: self.loginsKey, files: self.files)
+        BrowserProfile.loginsDBCreated = true
+        return db
```

Comment:
Looks like `lazy` properties are still unsafe

https://bugs.swift.org/browse/SR-1042

I'll investigate moving these into regular properties and de-singleton Profile.